### PR TITLE
Add io.github.HotShots

### PR DESCRIPTION
**Pull request by fastrizwaan**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

The software is unmaintained, but still very useful.

### CODE REVIEW

**Comment by fastrizwaan** [comment]
bot build io.github.HotShots

File: io.github.HotShots.desktop [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Version=2.2.0
```
Comment (bilelmoussaoui) [inline io.github.HotShots.desktop:2]:
```suggestion
```
That's the version of the specification, not the app version :-)

File: io.github.HotShots.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>io.github.HotShots</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0</project_license>
+  <name>Flameshot</name>
```
Comment (bilelmoussaoui) [inline io.github.HotShots.metainfo.xml:6]: Flameshot or HotShots?

File: io.github.HotShots.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  - --socket=fallback-x11
+  - --device=dri
+  - --filesystem=xdg-pictures
+  - --share=network
+  
+modules:
+- name: HotShots
+  sources:
+  - type: git
+    url: https://github.com/obiwankennedy/HotShots
```
Comment (bilelmoussaoui) [inline io.github.HotShots.yml:19]: Should be a tagged release or a specific commit

File: io.github.HotShots.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  - type: file
+    path: io.github.HotShots.desktop
+  
+  - type: file
+    path: io.github.HotShots.metainfo.xml
```
Comment (bilelmoussaoui) [inline io.github.HotShots.yml:25]: Should be upstream if possible

File: io.github.HotShots.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+   - install -Dm644 /app/share/pixmaps/hotshots.png 
+                    /app/share/icons/hicolor/128x128/apps/io.github.HotShots.png
+   - install -Dm644 io.github.HotShots.desktop /app/share/applications/
+   
+   - mkdir -p /app/share/metainfo
+   - install -Dm644 io.github.HotShots.metainfo.xml /app/share/metainfo
+
+#  build-options:
+#    build-args:
+#    - "--share=network"
```
Comment (bilelmoussaoui) [inline io.github.HotShots.yml:44]: Are those needed?

File: io.github.HotShots.metainfo.xml (hunk 2 above)
Comment (fastrizwaan) [reply io.github.HotShots.metainfo.xml:6]: > Flameshot or HotShots?

HotShots indeed, used Flameshot's appdata.

**Comment by bilelmoussaoui** [comment]
>     * [x]  I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
 
I would appreciate if this is correctly set on your request here.

**Comment by bilelmoussaoui** [comment]
bot, build io.github.HotShots

File: io.github.HotShots.yml (hunk 4 above)
Comment (fastrizwaan) [reply io.github.HotShots.yml:25]: requested, https://github.com/obiwankennedy/HotShots/issues/8

File: io.github.HotShots.yml (hunk 5 above)
Comment (fastrizwaan) [reply io.github.HotShots.yml]: Don't know, hence commented, thought network is required for git

**Comment by fastrizwaan** [comment]
> > ```
> > * [x]  I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
> > ```
> 
> I would appreciate if this is correctly set on your request here.

https://github.com/obiwankennedy/HotShots/issues/9
Requested.

File: io.github.HotShots.yml (hunk 3 above)
Comment (fastrizwaan) [reply io.github.HotShots.yml:19]: thank you.

**Comment by fastrizwaan** [comment]
bot, build io.github.HotShots

**Comment by fastrizwaan** [comment]
Kindly review

**Comment by fastrizwaan** [comment]
bot, build io.github.HotShots

File: io.github.HotShots.yml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+sdk: org.kde.Sdk
+runtime-version: '5.15'
+
+command: hotshots
+
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --device=dri
+  - --filesystem=~/xdg-pictures;`/xdg-documents;
```
Comment (barthalion) [inline io.github.HotShots.yml:12]:
```suggestion
  - --filesystem=xdg-pictures
  - --filesystem=xdg-documents
```

**Comment by fastrizwaan** [comment]
bot, build io.github.HotShots

**Comment by fastrizwaan** [comment]
Kindly Review!

**Comment by fastrizwaan** [comment]
Hi, any update on this?

**Comment by nedrichards** [comment]
This doesn't save files for me, I just get "error saving file" no matter where I try and save it (including after granting --filesystem=home permissions as an override.

**Comment by fastrizwaan** [comment]
> This doesn't save files for me, I just get "error saving file" no matter where I try and save it (including after granting --filesystem=home permissions as an override.

Yes, it requires file extension to save. Choose bmp and type temp.bmp, or choose png and type hello.png it will save. I have raised the issue. https://github.com/obiwankennedy/HotShots/issues/11

**Comment by fastrizwaan** [comment]
> This doesn't save files for me, I just get "error saving file" no matter where I try and save it (including after granting --filesystem=home permissions as an override.
Could it be that sandbox is interfering with file save dialog of Qt?
The application behaves differently when run on host system, it shows proper file name, whereas in flatpak it does not
Peek 2021-03-10 23-46

**Comment by barthalion** [comment]
bot, build io.github.HotShots

File: io.github.HotShots.metainfo.xml [hunk 7]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>io.github.HotShots</id>
```
Comment (barthalion) [inline io.github.HotShots.metainfo.xml:3]: This is not a proper ID. It should be `io.github.obiwankennedy.HotShots`.

**Review by barthalion — COMMENTED** [review]
The app works for me, just the ID is wrong.

**Comment by barthalion** [comment]
bot, build io.github.obiwankennedy.HotShots

**Comment by barthalion** [comment]
The manifest needs renaming too.

**Comment by fastrizwaan** [comment]
> The manifest needs renaming too.

Done

**Comment by fastrizwaan** [comment]
bot, build io.github.obiwankennedy.HotShots

**Comment by barthalion** [comment]
bot, build io.github.obiwankennedy.HotShots

**Pull request closed without merging**